xdmod/_http_requester.py

```diff
@@ -0,0 +1,160 @@
+        if post_fields:
+            post_fields['Bearer'] = self.__api_token
+            self.__crl.setopt(pycurl.POSTFIELDS, urlencode(post_fields))
+        else:
+            url += '&' if '?' in url else '?'
+            url += 'Bearer=' + self.__api_token
+        self.__crl.setopt(pycurl.URL, url)
+        self.__crl.setopt(pycurl.HTTPHEADER, self.__headers)
+        buffer = io.BytesIO()
+        self.__crl.setopt(pycurl.WRITEDATA, buffer)
+        try:
+            self.__crl.perform()
+        except pycurl.error as e:
+            code, msg = e.args
+            if code == pycurl.E_URL_MALFORMAT:
+                msg = 'Malformed URL.'
+            raise RuntimeError(msg) from None
+        response = buffer.getvalue().decode()
+        code = self.__crl.getinfo(pycurl.RESPONSE_CODE)
+        if code != 200:
+            msg = ''
+            try:
+                response_json = json.loads(response)
+                msg = ': ' + response_json['message']
+            except json.JSONDecodeError:
+                pass
+            if code == 401:
+                msg = (
+                    ': Make sure XDMOD_API_TOKEN is set to a valid API token.'
+                )
+            raise RuntimeError('Error ' + str(code) + msg) from None
+        return response
+
+    def __get_data_post_fields(self, params):
+        post_fields = {
+            'operation': 'get_data',
+            'start_date': params['start_date'],
+            'end_date': params['end_date'],
+            'realm': params['realm'],
+            'statistic': params['metric'],
+            'group_by': params['dimension'],
+            'dataset_type': (
+                'timeseries' if params['timeseries'] else 'aggregate'
+            ),
+            'aggregation_unit': params['aggregation_unit'],
+            'format': 'csv',
+        }
+        for dimension in params['filters']:
+            post_fields[dimension + '_filter'] = ','.join(
+                params['filters'][dimension]
+            )
+        return post_fields
+
+    def __get_raw_data_url_params(self, params):
+        results = {
+            'realm': params['realm'],
+            'start_date': params['start_date'],
+            'end_date': params['end_date'],
+        }
+        if (params['fields']):
+            results['fields'] = ','.join(params['fields'])
+        if (params['filters']):
+            results['filter_keys'] = ','.join(params['filters'])
+            for dimension in params['filters']:
+                results[dimension + '_filter'] = ','.join(
+                    params['filters'][dimension]
+                )
+        return urlencode(results)
+
+    def __get_raw_data_limit(self):
+        if self.__raw_data_limit is None:
+            response = self._request_json('/rest/v1/warehouse/raw-data/limit')
+            self.__raw_data_limit = int(response['data'])
+        return self.__raw_data_limit
+
+    def __get_environment_variable(self, name):
```

Comment:
Unused function? I can't see where this is called.

Reply:
Indeed, will remove it in the next commit.